### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* Focus on UXP + Universal Marketplace
* Abstractions documentation update
* Switch to declarative Configuration installation, avoid kubectl plugin usage
* Various additional fixes

Signed-off-by: Yury Tsarev <yury@upbound.io>

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

By following the guide

### CODE REVIEW

**Review by stevendborrelli — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 This repository contains a reference AWS Platform
 [Configuration](https://crossplane.io/docs/v1.9/getting-started/create-configuration.html)
-for use as a starting point in [Upbound Cloud](https://upbound.io) or
+for use as a starting point in [Universal Marketplace](https://marketplace.upbound.io/) and
```
Comment (jeanduplessis) [inline README.md:5]: Is it possibly misleading to say "for use" in Universal Marketplace? You're not really using it in there, merely accessing it from there right?

File: README.md [hunk 2]
```diff
@@ -79,45 +86,14 @@ Install the following command line tools:
...
-* `kubectl crossplane extension` (optional)
-
-  ```console
-  curl -sL https://raw.githubusercontent.com/crossplane/crossplane/master/install.sh | sh
-  cp kubectl-crossplane /usr/local/bin
-  ```
-
 ## Quick Start
 
 ### Platform Ops/SRE: Run your own internal cloud platform
```
Comment (jeanduplessis) [inline README.md:91]: This heading seems out of place in the overall content hierarchy now with the removal of the install instructions?

File: README.md [hunk 3]
```diff
@@ -233,47 +209,65 @@ kubectl delete providers.pkg.crossplane.io provider-aws
...
-    * `DBSubnetGroup`
+  * [definition.yaml](package/database/postgres/definition.yaml)
+  * [composition.yaml](package/database/postgres/composition.yaml) includes:
+    * `Instance`
+    * `SubnetGroup`
 
 ## Customize for your Organization
 
+You can customize this platform reference as much as you like and use it as
+a foundation for your very own Configuration building.
```
Comment (jeanduplessis) [inline README.md:259]:
```suggestion
You can customize this platform reference as much as you like and use it as
a foundation for building your very own Configuration.
```

**Review by jeanduplessis — APPROVED** [review]

**Comment by jeanduplessis** [comment]
@ytsarev LGTM, just a few questions and one grammar change proposal, nothing blocking.

**Comment by ytsarev** [comment]
@jeanduplessis thanks a lot for the review! I've updated the wording and layout, could you please check the fixes?

**Review by jeanduplessis — APPROVED** [review]
LGTM. Let's 🚢 it! 🚀

**Pull request merged**